bridges/ethereum/src/actors/block_ticker.rs

```diff
@@ -0,0 +1,97 @@
+//! Periodically check for blocks in block relay
+
+use crate::{config::Config, eth::EthState};
+use async_jsonrpc_client::futures::Stream;
+use futures::{future::Either, sink::Sink};
+use log::*;
+use std::{
+    collections::HashMap,
+    sync::Arc,
+    time::{Duration, Instant},
+};
+use tokio::{sync::mpsc, sync::oneshot, timer::Interval};
+use web3::{
+    contract,
+    futures::{future, Future},
+    types::U256,
+};
+
+/// Periodically check for blocks in block relay
+pub fn block_ticker(
+    config: Arc<Config>,
+    eth_state: Arc<EthState>,
```

Comment:
Tip: I would change the type of `config` to just references `config: &Config then in the caller just pass `block_ticker(&config...)`

Reply:
Unfortunately I cannot do that because the lifetime of config must be valid for the 'static lifetime, as the future returned by `block_ticker` can outlive the config.

Reply:
Yes you can baby, just copy `config.eth_account` (which is a `Copy` type) you need into the function scope and move that into the future:

```rust
/// Periodically check for blocks in block relay
pub fn block_ticker(
    config: &Config,
    eth_state: Arc<EthState>,
) -> (
    mpsc::Sender<(U256, oneshot::Sender<()>)>,
    impl Future<Item = (), Error = ()>,
) {
    let eth_account = config.eth_account;
    ...
```

Reply:
True, thanks!